spot_driver/spot_driver/spot_ros2.py

```diff
@@ -2793,17 +2795,32 @@ def step(self) -> None:
             self.mobility_params_pub.publish(mobility_params_msg)
 
     def destroy_node(self) -> None:
-        self.get_logger().info("Shutting down ROS driver for Spot")
-        if self.spot_wrapper is not None:
-            self.spot_wrapper.sit()
+        # self.get_logger().info("Shutting down ROS driver for Spot")
+        self.node_rate.sleep()
         if self.spot_wrapper is not None:
             self.spot_wrapper.disconnect()
+            self.node_rate.sleep()
         super().destroy_node()
 
 
 @ros_process.main(prebaked=False)
 def main(args: Optional[List[str]] = None) -> None:
-    ros_process.spin(SpotROS)
+    try:
+        ros_process.spin(SpotROS)
+
+    except KeyboardInterrupt:
+        print(COLOR_YELLOW + "User interrupt!" + COLOR_END)
+        ros_process.node().spot_wrapper.stop()
+
+    except (RetryableUnavailableError, InternalServerError, ConnectionError, 
+            rclpy.Exceptions.RCLError, ProxyConnectionError, UnableToConnectToRobotError) as e:
+        print(COLOR_RED + f"Connection error: {str(e)}" + COLOR_END)
+
+    finally:
+        time.sleep(2.0)
+        ros_process.node().destroy_node()
+        time.sleep(5.0)
```

Comment:
Why so much time sleeping here?

Reply:
I was experimenting with the ros shutdown process previously (i thought i had left this PR as a draft). Was trying to test if it was timing mismatch ros-type issues i was seeing or something else